Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraintPalmUp.cs

```diff
@@ -163,5 +211,193 @@ protected override bool IsValidController(IMixedRealityController controller)
+                    }
+                }
+            }
+
+            // Check if the palm angle meets the prescribed threshold
+            return palmCameraAngle < facingCameraTrackingThreshold;
+        }
+
+        /// <summary>
+        /// Checks to see if the user is currently gazing at the activation point; it first attempts to do so 
+        /// using eyegaze, and then falls back to head-based gaze if eyegaze isn't available for use.
+        /// </summary>
+        /// <param name="jointedHand"></param>
+        /// <returns></returns>
+        private bool IsUserGazeMeetingThresholdRequirements(IMixedRealityHand jointedHand)
+        {
+            Ray gazeRay;
+
+            bool usedEyeGaze = InputRayUtils.TryGetRay(InputSourceType.Eyes, Handedness.Any, out gazeRay);
+
+            if (usedEyeGaze || InputRayUtils.TryGetRay(InputSourceType.Head, Handedness.Any, out gazeRay))
+            {
+                // Define the activation point as a vector between the wrist and pinky knuckle; then cast it against the plane to get a smooth location
+                Vector3 activationPoint;
+                Plane handPlane;
+                float distanceToHandPlane;
+
+                // If we can generate the handplane/are able to set an activation point on it, and then are able to raycast against it
+                if (GenerateHandPlaneAndActivationPoint(jointedHand, out handPlane, out activationPoint) && 
+                    handPlane.Raycast(gazeRay, out distanceToHandPlane))
+                {
+                        // Now that we know the dist to the plane, create a vector at that point
+                        Vector3 gazePosOnPlane = gazeRay.origin + gazeRay.direction.normalized * distanceToHandPlane;
+                        Vector3 PlanePos = handPlane.ClosestPointOnPlane(gazePosOnPlane);
+                        float gazePosDistToActivationPosition = (activationPoint - PlanePos).sqrMagnitude;
+                        float gazeActivationThreshold = usedEyeGaze ? eyeGazeProximityThreshold : headGazeProximityThreshold;
+                        bool gazeActivated = gazeActivationAlreadyTriggered = (gazePosDistToActivationPosition < gazeActivationThreshold);
+
+                        return gazeActivated;
+                    }
+                }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Coroutine function called by the ManipulationHandler of the attached object whenever the object is done 
+        /// being manipulated by the user. This triggers a coroutine that checks to see whether the object should 
+        /// reattach to the hand.
+        /// </summary>
+        public void StartWorldLockReattachCheckCorotine()
+        {
+            StartCoroutine(WorldLockedReattachCheck());
+        }
+
+        /// <summary>
+        /// This function attenpts to generate a hand plane based on the wrist, index knuckle and pinky knuckle joints present in the hand.
```

Comment:
```suggestion
        /// This function attempts to generate a hand plane based on the wrist, index knuckle and pinky knuckle joints present in the hand.
```

You can install VSCode spell checker extensions to catch these spelling errors.